service/src/main/java/com/codeforcommunity/processor/ProtectedReportProcessorImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.codeforcommunity.processor;
+
+import com.codeforcommunity.api.IProtectedReportProcessor;
+import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.report.AdoptedSite;
+import com.codeforcommunity.dto.report.GetAdoptionReportResponse;
+import com.codeforcommunity.dto.report.GetStewardshipReportResponse;
+import com.codeforcommunity.dto.report.Stewardship;
+import com.codeforcommunity.enums.PrivilegeLevel;
+import com.codeforcommunity.exceptions.AuthException;
+import org.jooq.DSLContext;
+import java.util.List;
+
+import static org.jooq.generated.Tables.ADOPTED_SITES;
+import static org.jooq.generated.Tables.NEIGHBORHOODS;
+import static org.jooq.generated.Tables.SITES;
+import static org.jooq.generated.Tables.STEWARDSHIP;
+import static org.jooq.generated.tables.Users.USERS;
+import static org.jooq.impl.DSL.concat;
+import static org.jooq.impl.DSL.val;
+import static org.jooq.impl.DSL.count;
+
+public class ProtectedReportProcessorImpl implements IProtectedReportProcessor {
+
+    private final DSLContext db;
+
+    public ProtectedReportProcessorImpl(DSLContext db) {
+        this.db = db;
+    }
+
+    /**
+     * Throws an exception if the user is not an admin or super admin.
+     *
+     * @param level the privilege level of the user calling the route
+     */
+    void isAdminCheck(PrivilegeLevel level) {
+        if (!(level.equals(PrivilegeLevel.ADMIN) || level.equals(PrivilegeLevel.SUPER_ADMIN))) {
+            throw new AuthException("User does not have the required privilege level.");
```

Comment:
⛏️ Might be more helpful for the user if the message specified they had to be an admin.